src/app/parts-editor/services/parts-data.service.ts

```diff
@@ -5,12 +5,22 @@ import { BehaviorSubject, Observable, Subject } from 'rxjs';
 import { PART_FAILURES, PartFailure } from '../models/part-failure';
 import { PartEditor, PartWorkbook } from '../models/editor';
 import { Cell } from '../models/cell';
+import { DialogDatabaseComponent } from '../components/dialog-database/dialog-database.component';
+import { MatDialog } from '@angular/material/dialog';
 
 export interface EditorsUiState {
   activeIndex: number | null;
   openedEditors: PartEditor[];
 }
 
+export interface UpdatedPart {
```

Comment:
If its not strongly connected with this service it should be moved to separate file in models directory. Also probably it would be better to rename as PartToUpdate. Another approach for that crud models is to use crud operation as suffix ex PartUpdate, PartCreate etc